pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -760,3 +775,34 @@ func ipAddrIsValid(ipAddr net.IP) bool {
 	}
 	return true
 }
+
+func getNIProbeMetric(ctx *zedrouterContext, netstatus *types.NetworkInstanceStatus) types.ProbeMetrics {
+	var metrics types.ProbeMetrics
+	remoteURL := getSystemURL()
+	probeMutex.Lock()
```

Comment:
Better to put a defer ptobrMutrz.Unlock() here instead of just before the return.

Reply:
will do.